### PR TITLE
Granular CRN for shared processor pool data sources and resources

### PR DESCRIPTION
**Pull request by Alexander-Kita**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

pi_shared_processor_pool (d):

```
=== RUN   TestAccIBMPIPISharedProcessorPoolDataSource_basic
--- PASS: TestAccIBMPIPISharedProcessorPoolDataSource_basic (26.84s)
PASS
```

pi_shared_processor_pools (d):

```
=== RUN   TestAccIBMPISharedProcessorPoolsDataSourceBasic
--- PASS: TestAccIBMPISharedProcessorPoolsDataSourceBasic (11.94s)
PASS
```

pi_shared_processor_pool (r):

```
=== RUN   TestAccIBMPISPPbasic
--- PASS: TestAccIBMPISPPbasic (49.45s)
PASS
```

```
=== RUN   TestAccIBMPISPPUserTags
--- PASS: TestAccIBMPISPPUserTags (96.87s)
PASS
```

### CODE REVIEW

**Comment by Alexander-Kita** [comment]
@yussufsh, @hkantare Can you please review this PR which part of power Q3

**Review by ismirlia — APPROVED** [review]

**Review by ismirlia — APPROVED** [review]

**Review by ismirlia — APPROVED** [review]

**Review by ismirlia — DISMISSED** [review]

**Review by ismirlia — DISMISSED** [review]

File: ibm/service/power/resource_ibm_pi_shared_processor_pool.go [hunk 1]
```diff
@@ -265,6 +283,9 @@ func resourceIBMPISharedProcessorPoolRead(ctx context.Context, d *schema.Resourc
 	}
 
 	d.Set(Arg_CloudInstanceID, cloudInstanceID)
+	if response.SharedProcessorPool.Crn != "" {
+		d.Set(Attr_CRN, response.SharedProcessorPool.Crn)
+	}
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_shared_processor_pool.go:288]: This is duplicated, since you have that group with the tag

**Review by michaelkad — COMMENTED** [review]
Duplicate code

File: ibm/service/power/resource_ibm_pi_shared_processor_pool.go (hunk 1 above)
Comment (Alexander-Kita) [reply ibm/service/power/resource_ibm_pi_shared_processor_pool.go]: Fixed

**Review by ismirlia — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Review by yussufsh — APPROVED** [review]

**Pull request merged**